modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -205,6 +212,27 @@
    (driver.common/start-of-week-offset driver)
    (partial hsql/call (u/qualified-name ::mod))))
 
+(defmethod sql.qp/->honeysql [:oracle :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [expr          (sql.qp/->honeysql driver arg)
+        has-timezone? (hx/is-of-type? expr #"timestamp(\(\d\))? with time zone")]
+    (when (and has-timezone? source-timezone)
+      (throw (ex-info (tru "`timestamp with time zone` columns shouldn''t have a `source timezone`")
+                      {:target-timezone target-timezone
+                       :source-timezone source-timezone
+                       :type            qp.error-type/invalid-parameter})))
+    (let [source-timezone (or source-timezone (qp.timezone/results-timezone-id))
+          hsql-from-tz    (partial hsql/call :from_tz)]
+      (cond-> expr
+        (not has-timezone?)
+        (hsql-from-tz source-timezone)
```

Comment:
Why not just
```suggestion
        (hsql/call :from_tz source-timezone)
```
shorter + easier to understand

Reply:
It's under a threading first, so this will not work.

Reply:
I just realised `target-timezone` can never be falsy too. Would this make it clearer?
```suggestion
      (-> (if has-timezone? expr (hsql/call :from_tz expr source-timezone)
          (hx/->AtTimeZone target-timezone)
          hx/->timestamp))))
```